KuringApp/KuringApp/Setting/AppIconSelectView.swift

```diff
@@ -0,0 +1,130 @@
+//
+//  AppIconSelectView.swift
+//  KuringApp
+//
+//  Created by 박성수 on 2023/09/27.
+//
+
+import SwiftUI
+import ComposableArchitecture
+
+enum KuringIcon: String, CaseIterable {
+    case kuring_app
+    case kuring_app_classic
+    case kuring_app_blueprint
+    case kuring_app_sketch
+    
+    var korValue: String {
+        switch self {
+        case .kuring_app: return "쿠링 기본"
+        case .kuring_app_classic: return "쿠링 클래식"
+        case .kuring_app_blueprint: return "쿠링 블루프린트"
+        case .kuring_app_sketch: return "쿠링 스케치"
+        }
+    }
+}
+
+struct AppIconSelect: Reducer {
+    struct State: Equatable {
+        var appIcons: [KuringIcon] = KuringIcon.allCases
+        var selectedIcon: KuringIcon?
+    }
+    
+    enum Action: Equatable {
+        case appIconButtonTapped(tappedValue: KuringIcon)
+        case saveButtonTapped
+        
+        // MARK: To SettingView
+        case delegate(Delegate)
+        enum Delegate: Equatable {
+            case alternativeAppIconSave(KuringIcon)
+        }
+    }
+    
+    var body: some ReducerOf<Self> {
+        Reduce { state, action in
+            switch action {
+            case let .appIconButtonTapped(tappedValue: tappedValue):
+                state.selectedIcon = tappedValue
```

Comment:
```suggestion
            case let appIconSelected(id):
                guard let icon = state.appIcons[id: id] else {
                    return .none
                }state.selectedIcon = icon
```
아니면 selectedIconID 같이 id값만 들고 있다가 save 누르면 그때 아이콘 가져와서 업데이트 해도 될거 같습니다